torchani/cuaev/cuaev.cpp

```diff
@@ -0,0 +1,198 @@
+    int64_t angular_length_aligned_,
+    int64_t ncenter_atoms_,
+    Tensor coordinates_t_,
+    Tensor species_t_)
+    : aev_t(aev_t_),
+      tensor_Rij(tensor_Rij_),
+      tensor_radialRij(tensor_radialRij_),
+      tensor_angularRij(tensor_angularRij_),
+      total_natom_pairs(total_natom_pairs_),
+      nRadialRij(nRadialRij_),
+      nAngularRij(nAngularRij_),
+      tensor_centralAtom(tensor_centralAtom_),
+      tensor_numPairsPerCenterAtom(tensor_numPairsPerCenterAtom_),
+      tensor_centerAtomStartIdx(tensor_centerAtomStartIdx_),
+      maxnbrs_per_atom_aligned(maxnbrs_per_atom_aligned_),
+      angular_length_aligned(angular_length_aligned_),
+      ncenter_atoms(ncenter_atoms_),
+      coordinates_t(coordinates_t_),
+      species_t(species_t_) {}
+
+Result::Result(tensor_list tensors)
+    : aev_t(tensors[0]),
+      tensor_Rij(tensors[1]),
+      tensor_radialRij(tensors[2]),
+      tensor_angularRij(tensors[3]),
+      total_natom_pairs(tensors[4].item<int>()),
+      nRadialRij(tensors[5].item<int>()),
+      nAngularRij(tensors[6].item<int>()),
+      tensor_centralAtom(tensors[7]),
+      tensor_numPairsPerCenterAtom(tensors[8]),
+      tensor_centerAtomStartIdx(tensors[9]),
+      maxnbrs_per_atom_aligned(tensors[10].item<int>()),
+      angular_length_aligned(tensors[11].item<int>()),
+      ncenter_atoms(tensors[12].item<int>()),
+      coordinates_t(tensors[13]),
+      species_t(tensors[14]) {}
+
+tensor_list Result::to_list() {
+  return {aev_t,
+          tensor_Rij,
+          tensor_radialRij,
+          tensor_angularRij,
+          torch::tensor(total_natom_pairs),
+          torch::tensor(nRadialRij),
+          torch::tensor(nAngularRij),
+          tensor_centralAtom,
+          tensor_numPairsPerCenterAtom,
+          tensor_centerAtomStartIdx,
+          torch::tensor(maxnbrs_per_atom_aligned),
+          torch::tensor(angular_length_aligned),
+          torch::tensor(ncenter_atoms),
+          coordinates_t,
+          species_t};
+}
+
+void Result::release() {
+  aev_t = Tensor();
+  tensor_Rij = Tensor();
+  tensor_radialRij = Tensor();
+  tensor_angularRij = Tensor();
+  tensor_centralAtom = Tensor();
+  tensor_numPairsPerCenterAtom = Tensor();
+  tensor_centerAtomStartIdx = Tensor();
+  coordinates_t = Tensor();
+  species_t = Tensor();
+}
+
+CuaevComputer::CuaevComputer(
+    double Rcr,
+    double Rca,
+    const Tensor& EtaR_t,
+    const Tensor& ShfR_t,
+    const Tensor& EtaA_t,
+    const Tensor& Zeta_t,
+    const Tensor& ShfA_t,
+    const Tensor& ShfZ_t,
+    int64_t num_species)
+    : aev_params(Rcr, Rca, EtaR_t, ShfR_t, EtaA_t, Zeta_t, ShfA_t, ShfZ_t, num_species) {}
+
+Tensor CuaevDoubleAutograd::forward(
+    AutogradContext* ctx,
+    Tensor grad_e_aev,
+    const torch::intrusive_ptr<CuaevComputer>& cuaev_computer,
+    tensor_list result_tensors) {
+  Result result(result_tensors);
+  Tensor grad_coord = cuaev_computer->backward(grad_e_aev, result);
```

Comment:
I think we can just do
```suggestion
  Tensor grad_coord = cuaev_computer->backward(grad_e_aev, result_tensors);
```
and c++ will automatically convert tensor_list to Result?